intern/resource/dvid/tests/test_data_instance.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2020 The Johns Hopkins University Applied Physics Laboratory
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+from intern.resource.dvid.resource import DataInstanceResource
+
+class TestDataInstanceResource(unittest.TestCase):
+    def setUp(self):
+
+        UUID = "822524777d3048b8bd520043f90c1d28"
+        ALIAS = "grayscale"
+        
+        self.name = "some_data_instance"
+        self.uuid = UUID
+        self.alias = ALIAS
+
+        self.data_instance_master = DataInstanceResource(self.name, UUID = self.uuid, datatype="uint8")
+
+    def test_init_status_false(self):
+        data_instance = DataInstanceResource(self.name, UUID = self.uuid)
+        self.assertFalse(data_instance.cutout_ready)
+
+    def test_channel_defaults_to_uint8(self):
+        data_instance = DataInstanceResource(self.name, UUID = self.uuid)
+        self.assertEqual('uint8', data_instance.datatype)
+
+    def test_channel_defaults_to_uint8blk_type(self):
+        data_instance = DataInstanceResource(self.name, UUID = self.uuid)
+        self.assertEqual('uint8blk', data_instance._type)
+    
+    def test_setting_datatype_means_cutout_ready_at_construction(self):
+        data_instance = DataInstanceResource(self.name, UUID = self.uuid, datatype="uint8")
+        self.assertTrue(data_instance.cutout_ready)
+
+    def test_setting_datatype_means_cutout_ready(self):
+        data_instance = DataInstanceResource(self.name, UUID = self.uuid)
+
+        # First check should be false. 
+        self.assertFalse(data_instance.cutout_ready)
+
+        #Set the datatype now
+        data_instance.datatype = 'uint16'
+
+        # Now we should be ready
+        self.assertTrue(data_instance.cutout_ready)
+
+    def test_valid_volume(self):
+        self.assertTrue(self.data_instance_master.valid_volume())
+
+    def test_validate_datatype_uint8(self):
```

Comment:
I didn't know about `subTest()` when I did the same thing for Boss. It'd save a lot of typing.

Reply:
haha nice, I did the same things here. I think it's fine though